assimilation_code/modules/assimilation/assim_tools_mod.f90

```diff
@@ -78,13 +78,17 @@ module assim_tools_mod
 
 use algorithm_info_mod, only : probit_dist_info, obs_inc_info, EAKF, ENKF, &
                                BOUNDED_NORMAL_RHF, UNBOUNDED_RHF, GAMMA_FILTER, &
-                               KERNEL, OBS_PARTICLE
+                               KERNEL, OBS_PARTICLE, KERNEL_QCEF
 
 use gamma_distribution_mod, only : gamma_cdf, inv_gamma_cdf, gamma_mn_var_to_shape_scale, &
                                    gamma_gamma_prod
 
 use bnrh_distribution_mod, only   :  inv_bnrh_cdf, bnrh_cdf, inv_bnrh_cdf_like
 
+use kde_distribution_mod, only : kde_cdf_params, inv_kde_cdf_params, obs_dist_types,      &
```

Comment:
this is a style naming convention, we use _type for derived types. @hkershaw This would be quite a few lines of code to change, so if we switch this we'll do a global search & replace single commit.

Reply:
OK, I will hold off on this for now.